app/controllers/sessions_controller.rb

```diff
@@ -1,2 +1,27 @@
 class SessionsController < ApplicationController
+  before_action :require_login, except: [:new, :create]
+  skip_before_filter :verify_authenticity_token, only: [:create]
+
+  def new
+  end
```

Comment:
You can put empty methods like this on one line with `def new; end`. The one line version is preferred in the [style guide](https://github.com/bbatsov/ruby-style-guide#single-line-classes).